scripts/preview_outcome.py

```diff
@@ -14,7 +14,12 @@ def main(book:str, outcome:str, amount:int = 20):
     sage_library_path = Path("source", book, "sage")
     bank_root = ET.parse(exercise_path/"bank.xml")
     search_string = f'.//*[name()="slug" and text()="{outcome}"]/./..'
-    outcome_ele = bank_root.xpath(search_string)[0]
+    element_list = bank_root.xpath(search_string)
+    if element_list == []:
+        error_message = f"{outcome} is not an outcome defined in {exercise_path}/bank.xml"
+        raise ValueError(error_message)
+    else:
+        outcome_ele = element_list[0]
```

Comment:
```suggestion
    try:
        outcome_ele = bank_root.xpath(search_string)[0]
    except IndexError:
        error_message = f"{outcome} is not an outcome defined in {exercise_path}/bank.xml"
        raise ValueError(error_message)
```

Bit more Pythnonic to try/except in this situation.